go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -333,17 +333,19 @@ func (throttler *Throttler) WatchSrvKeyspaceCallback(srvks *topodatapb.SrvKeyspa
 // This may cause the throttler to be enabled/disabled, and of course it affects the throttling query/threshold.
 // Note: you should be holding the initMutex when calling this function.
 func (throttler *Throttler) applyThrottlerConfig(ctx context.Context, throttlerConfig *topodatapb.ThrottlerConfig) {
-	if !throttlerConfigViaTopo {
-		return
-	}
 	log.Infof("Throttler: applying topo config: %+v", throttlerConfig)
 	if throttlerConfig.CustomQuery == "" {
 		throttler.metricsQuery.Store(sqlparser.BuildParsedQuery(defaultReplicationLagQuery, sidecardb.GetIdentifier()).Query)
 	} else {
 		throttler.metricsQuery.Store(throttlerConfig.CustomQuery)
 	}
 	throttler.StoreMetricsThreshold(throttlerConfig.Threshold)
-	throttlerCheckAsCheckSelf = throttlerConfig.CheckAsCheckSelf
+	if throttlerConfig.CheckAsCheckSelf {
+		atomic.StoreInt64(&throttler.checkAsCheckSelf, 1)
+	} else {
+		atomic.StoreInt64(&throttler.checkAsCheckSelf, 0)
+	}
```

Comment:
Any reason not to start leveraging atomic.Bool for things like this, since we're modifying it?

Reply:
done